cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileMetricScheduledService.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.profile;
+
+import co.cask.cdap.api.metrics.MetricsCollectionService;
+import co.cask.cdap.api.metrics.MetricsContext;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.id.ProfileId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import org.apache.twill.common.Threads;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Profile metrics schedule service which will emit metrics about profile, by default it will emit per 60 seconds
+ */
+public class ProfileMetricScheduledService extends AbstractScheduledService {
+  private static final int DEFAULT_INTERVAL_SECS = 60;
+
+  private final MetricsContext metricsContext;
+  private final long intervalSecs;
+  private final ProgramRunId programRunId;
+  private final int numNodes;
+
+  private ScheduledExecutorService executor;
+
+  public ProfileMetricScheduledService(MetricsCollectionService metricsCollectionService, ProgramRunId programRunId,
+                                       ProfileId profileId, int numNodes) {
+    this(metricsCollectionService, programRunId, profileId, numNodes, DEFAULT_INTERVAL_SECS);
+  }
+
+  public ProfileMetricScheduledService(MetricsCollectionService metricsCollectionService, ProgramRunId programRunId,
+                                       ProfileId profileId, int numNodes, long intervalSecs) {
+    this.metricsContext = getMetricsContextForProfile(metricsCollectionService, programRunId, profileId);
+    this.numNodes = numNodes;
+    this.programRunId = programRunId;
+    this.intervalSecs = intervalSecs;
+  }
+
+  @Override
+  protected void runOneIteration() throws Exception {
+    long value = intervalSecs * numNodes / 60L;
```

Comment:
if intervalSecs is less than 60, this can be 0 for single node clusters. We should make sure intervalSecs is at least 60, or take minutes instead of seconds in the constructor